File: Algorithms/739.Daily-Temperatures/daily_temperatures.go
func dailyTemperatures(T []int) []int {
    length := len(T)
    res := make([]int, length)
    stack := make([]int, 0)
    
    for i := 0; i < length; i++ {
        slen := len(stack)
        for slen > 0 && T[i] > T[stack[slen - 1]] {
            res[stack[slen - 1]] = i - stack[slen - 1]
            stack = stack[:slen - 1]
            slen -= 1
        }
        stack = append(stack, i)
    }
    
    return res        
}

File: Algorithms/739.Daily-Temperatures/daily_temperatures.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        temp = []        
        length = len(T)
        res = [0] * length
        
        temp.append(0)
        for i in range(length):
            while temp and T[i] > T[temp[-1]]:
                idx = temp.pop()
                res[idx] = i - idx
            temp.append(i)
        return res
